polyfills/Array/from/polyfill.js

```diff
@@ -12,22 +11,18 @@
 		if (typeof arraylike.next === 'function') {
 			while (!done) {
 				iterableResponse = arraylike.next();
-				if (
-					iterableResponse.hasOwnProperty('value') &&
-					iterableResponse.hasOwnProperty('done')
-				) {
+				if (typeof iterableResponse.done === 'boolean') {
 					if (iterableResponse.done === true) {
 						done = true;
 						break;
-
-					// handle the case where the done value is not Boolean
-					} else if (iterableResponse.done !== false) {
+					}
+					//was using hasownProperty but changed as 'value' property might be inherited through prototype chain and could still be a valid iterable response
```

Comment:
Are you sure about this? I think the Iterator Result Object does not have a prototype and always has a `done` property but may not have a `value` property -- https://tc39.github.io/ecma262/#table-55